File: 3697-compute-decimal-representation/3697-compute-decimal-representation.cpp
class Solution {
public:
    vector<int> decimalRepresentation(int n) {
        vector<int> ans;
        long long power = 1;
        while(n > 0){
            long long d = n % 10;
            if(d > 0){
                long long temp = d * power;
                ans.push_back(temp);
            }
            n = n / 10;
            power = power * 10;    
        }
        reverse(ans.begin(), ans.end()); 
        return ans;
    }
};

File: 3697-compute-decimal-representation/3697-compute-decimal-representation.py
class Solution:
    def decimalRepresentation(self, n: int) -> List[int]:
        ans = []
        power = 1
        while n > 0:
            d = n % 10
            if d > 0:
                temp = d * power
                ans.append(temp)
            n = n // 10
            power = power * 10
        return ans[::-1]    
                
                